kafka-streams-samples/kafka-streams-recoverable/src/test/java/kafka/streams/record/recoverable/KafkaStreamsRecoverableSampleTest.java

```diff
@@ -0,0 +1,115 @@
+        consumerProps.put(ConsumerConfig.AUTO_OFFSET_RESET_CONFIG, "earliest");
+        DefaultKafkaConsumerFactory<String, String> cf = new DefaultKafkaConsumerFactory<>(consumerProps);
+        consumer = cf.createConsumer();
+        embeddedKafka.consumeFromEmbeddedTopics(consumer, "topic-in", "default-out", "custom-out");
+        System.setProperty("spring.cloud.stream.kafka.streams.binder.brokers", embeddedKafka.getBrokersAsString());
+    }
+
+    @AfterClass
+    public static void tearDown() {
+        consumer.close();
+        System.clearProperty("spring.cloud.stream.kafka.streams.binder.brokers");
+    }
+
+    @Test
+    public void testKafkaStreams() {
+        Map<String, Object> senderProps = KafkaTestUtils.producerProps(embeddedKafka);
+        DefaultKafkaProducerFactory<Integer, String> pf = new DefaultKafkaProducerFactory<>(senderProps);
+        try {
+            KafkaTemplate<Integer, String> template = new KafkaTemplate<>(pf, true);
+            template.setDefaultTopic("topic-in");
+            template.send(MessageBuilder.withPayload("1").build());
+            template.send(MessageBuilder.withPayload("2").build());
+            template.send(MessageBuilder.withPayload("3").build());
+            template.send(MessageBuilder.withPayload("4").build());
+            template.send(MessageBuilder.withPayload("5").build());
+            ConsumerRecords<String, String> consumerRecords = KafkaTestUtils.getRecords(consumer);
+            var topicIn = new ArrayList<ConsumerRecord<String, String>>();
+            var defaultOut = new ArrayList<ConsumerRecord<String, String>>();
+            var customOut = new ArrayList<ConsumerRecord<String, String>>();
+
+            for(var record : consumerRecords.records("topic-in")) {
+                topicIn.add(record);
+            }
+            for(var record : consumerRecords.records("default-out")) {
+                defaultOut.add(record);
+            }
+            for(var record : consumerRecords.records("custom-out")) {
+                customOut.add(record);
+            }
+
+            assertEquals(6, topicIn.size());
+            assertThat(defaultOut.size()).isLessThanOrEqualTo(2);
+            assertThat(customOut.size()).isLessThanOrEqualTo(2);
+
+            assertEquals(0, Integer.parseInt(topicIn.get(0).value()));
+            assertEquals(1, Integer.parseInt(topicIn.get(1).value()));
+            assertEquals(2, Integer.parseInt(topicIn.get(2).value()));
+            assertEquals(3, Integer.parseInt(topicIn.get(3).value()));
+            assertEquals(4, Integer.parseInt(topicIn.get(4).value()));
+            assertEquals(5, Integer.parseInt(topicIn.get(5).value()));
+
+            assertEquals(0, Integer.parseInt(defaultOut.get(0).value()));
+            try {
+                assertEquals(5, Integer.parseInt(defaultOut.get(1).value()));
+            } catch (IndexOutOfBoundsException e) {
```

Comment:
`catch` on new line.

Reply:
Updated in 391f70fbd7ddef62de22c216276e38b246969f9e